### ISSUE TITLE
Add the current theme to the DOM

### ISSUE THREAD

**Issue by samelhusseini** [google/blockly#3407]
**Is your feature request related to a problem? Please describe.**

It's currently not possible to add CSS that is only relevant for a specific theme used. Say you have a default theme, and a high contrast theme. If you wanted to add additional properties to your CSS that only applied to the high contrast theme, that wouldn't be possible.

**Describe the solution you'd like**

Simple solution, set the current theme name as a class on the injection div.

**Describe alternatives you've considered**

**Additional context**

**Comment by samelhusseini** [comment]
Fixed.

**Comment by BeksOmega** [comment]
I've been really dying to ask this, but I've honestly been scared to. Why did you decide to go with themes over css? Don't get me wrong, I love themes, but it just feels like css would be the first method to try. Is there maybe an IE thing you ran into that makes css unusable?

**Comment by samelhusseini** [comment]
Hmm. I'm not sure what you mean?
Do you mean why we originally used JS to set component colours as opposed to doing this from the start?

**Comment by BeksOmega** [comment]
> Hmm. I'm not sure what you mean?
> Do you mean why we originally used JS to set component colours as opposed to doing something like this from the start?
> ps: Always feel free to always ask questions!

Yeah, like go through and make sure all dom elements were tagged with classes, and then allow styling with those.

**Comment by samelhusseini** [comment]
Yeah they are, and what we've settled on is: 
- Use CSS for styling
- For commonly configured colours use the themes object (workspace, toolbox, flyout). This makes it easier for developers to create dynamic themes (such as high contrast) that changes block colours, category colours, and major components of Blockly.
- For more complex theme adjustments (such as hover), we fallback on using CSS with the theme selector. 

The downside of using CSS for colours is that it cannot be (easily) dynamically adjusted. I say easily because the caveat there is you could dump out the CSS from a div, and re-inject new CSS that has the updated colours.
Doing the same thing with CSS would mean selectors like this: 
```
.dark-theme .blocklySvg {
 ....
}
.dark-theme .blocklyToolboxDiv {
 ....
}
```
etc. 

The above has a couple of issues. In order to adjust the theme, developers need to know of our CSS selectors. eg: we have the following selector ``.blocklyTreeRow:not(.blocklyTreeSelected):hover``. First of all, blocklyTreeRow isn't a great name for toolbox category, and with a toolbox re-write we'll probably change those to better names. Second of all, if you are adjusting the theme, without full knowledge of our CSS, you might think to add: 
```
my-custom-theme .blocklyTreeRow {
background-color: #MYCOLOR;
}
```
Unintentionally you can create a selector that is more specific than another selector in our CSS. 
There are ways to avoid that, but again you would need to know the CSS in depth.

Another thing to point out is you end up with a maintenance nightmare. Say we change our CSS slightly, now everyone has to adjust theres to match.

Another another thing, is you have to ship a lot more CSS to adjust a theme.

The goal here is to make it easy for developers to use the library and be able to make colour adjustments without digging deep into how our CSS is structured. You could argue Web developers are already pretty familiar with this part, and that is partially why I added the theme to the injection div, to let you use CSS if you wanted, but for the developers that want to just configure things in JSON (options), this is the path for them. 


tl;dr, Using theme components, is basically the "easy" way to adjust the theme colours. For more complicated adjustments, we rely on CSS.
